engines/e_dasync.c

```diff
@@ -7,6 +7,13 @@
  * https://www.openssl.org/source/license.html
  */
 
+/*
+ * SHA-1 low level APIs are deprecated for public use, but still ok for
+ * internal use.  Note, that due to symbols not being exported, only the
+ * #defines can be accessed, in this case SHA_CBLOCK.
```

Comment:
Perhaps just define SHA_CBLOCK locally in this file rather than including deprecated.h?

Reply:
Added note that `sizeof(ctx)` is also required.